docs/cloud/satellites/byoc/tailscale.md

```diff
@@ -0,0 +1,25 @@
+# Configuring Tailscale for BYOC
+
+Tailscale is a super-simple VPN that is easy to set up, and works well with BYOC satellites. This page documents the required configuration within Tailscale to enable BYOC.
+
+## Requirements
+
+* Configure a [subnet router](https://tailscale.com/kb/1019/subnets) to provide access to satellites for users on the VPN. This is required because satellites never join a VPN directly, and may change IP/DNS addresses frequently.
+* [Configure Restricted Nameservers (Split DNS)](https://tailscale.com/kb/1054/dns?q=split+dns#nameservers) to resolve custom DNS names, as required by your cloud provider.
+* If you are running Earthly from within a Kubernetes pod, or GHA runner; you may need to make use of the userspace networking mode.
```

Comment:
```suggestion
* If you are running Earthly from within a Kubernetes pod, or GHA runner; you may need to make use of the [userspace networking mode](https://tailscale.com/kb/1112/userspace-networking).
```

Reply:
Should we advice them to use the official Github actions module from Tailscale in GHA?

Reply:
We could also include full working examples here of using Tailscale from Github actions and Kuberentes, based on our own internal testing. I can help with the examples if you'd like, since it's all in my branch right now.

Reply:
I would like to avoid reproducing the documentation as much as possible. If its clear and step-by-step on Tailscale's side, we should keep it there. If there are Earthly-specific tweaks, then we should document those more?